### DISCUSSION TITLE
Match `/@:username` url

### DISCUSSION THREAD

**Discussion by happylolonly** [remix-run/react-router#10545] [Q&A] [answered]
Hi, how to match `/@:username` url?

**Comment by kiliman** [comment, marked as answer]
With RR 6+, params include the _entire_ URL segment, so it doesn't support prefixes or any type of pattern match. You will need to use `/:username` and check for and strip the `@` yourself.

**Reply by happylolonly** [reply]
thanks

**Reply by FireworksX** [reply]
@kiliman Can you show me please how i should make separate. I have two routes **Match** and **Player**, how i can strip this pattern? I understand like within RootRoute i should make conditional operator for render one of inside components? But Match and Player routes has unique children, how i need define routes?
```ts
{
  path: '',
  element: <RouteElements.RootRoute />,
  children: [
    {
      path: `m-:${ROUTE_PARAMS.matchSlug}`,
      element: <RouteElements.MatchRoute />,
      children: [
        {
          path: 'live',
          element: <RouteElements.MatchLiveRoute />
        }
      ]
    },
    {
      path: `p-:${ROUTE_PARAMS.playerSlug}`,
      element: <RouteElements.MatchRoute />,
      children: [
        {
          path: 'statistics',
          element: <RouteElements.PlayerStatisticsRoute />
        }
      ]
    }
  ]
}
```

**Reply by kiliman** [reply]
I'm not sure you can... well, at least not easily. Is it possible to simply change your URL structure? So instead of `/m-:match/live` do `/m/:match/live` and same with player `/p/:player/statistics`.

**Reply by FireworksX** [reply]
I have site which already use pattern like `m-:slug`, and i can't change route names because of site had indexes from Google. I wanna migrate to `react-router v6`, but i feel frustration what lead router with 50k starts can't resolve this pattern. Maybe do you have some plans about that?
